### PR TITLE
Catch all exceptions in context.user_agent

### PR DESCRIPTION
**Pull request by jaimergp**
### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/master/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

Follow-up for #11415 - ignore `ImportError` too if the solver cannot be imported. Otherwise, `conda info` will fail (and it might be used for debugging purposes).

Actually, we might want to ignore _all_ exceptions in this block? 

@jezdez

### CODE REVIEW

**Comment by jezdez** [comment]
@jaimergp Yeah, I know it's a pretty big anti-pattern, but we should swallow exceptions to make user experience not a problem. It's not great, but better than risking stuff.

File: conda/base/context.py [hunk 1]
```diff
@@ -829,7 +829,7 @@ def user_agent(self):
             try:
                 # Solver.user_agent has to be a static or class method
                 user_agent_str += f" {_get_solver_class().user_agent()}"
-            except (AttributeError, TypeError):
+            except (AttributeError, TypeError, ImportError):
```
Comment (jezdez) [inline conda/base/context.py:832]:
```suggestion
            except Exception:
```

**Comment by jaimergp** [comment]
@jezdez - do we want to log the exception via `log.debug` or something? It could be painful to debug otherwise.

**Comment by jezdez** [comment]
> @jezdez - do we want to log the exception via `log.debug` or something? It could be painful to debug otherwise.

Yeah, good idea.

**Comment by beeankha** [comment]
@anaconda-issue-bot check

**Review by jezdez — APPROVED** [review]

**Pull request merged**